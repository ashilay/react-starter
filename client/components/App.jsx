import React from 'react';
import PropTypes from 'prop-types';

export default class App extends React.Component {
    getFullName() {
        return `${this.props.name} ${this.props.sername}`;
    }

    render() {
        return (
            <div style={{ textAlign: 'center' }}>
                <h1>Hello, {this.getFullName()}!</h1>
            </div>
        );
    }
}

App.propTypes = {
    name: PropTypes.string.isRequired,
    sername: PropTypes.string.isRequired,
};
