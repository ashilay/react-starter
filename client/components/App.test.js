import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

//Snapshot Testing
test('render Hellow World text', () => {
    const component = renderer.create(
        <App />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
