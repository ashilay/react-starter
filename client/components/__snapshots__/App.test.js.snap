// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Hellow World text 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h1>
    Hello World!
  </h1>
</div>
`;
